app/controllers/pages_controller.rb

```diff
@@ -1,6 +1,5 @@
 class PagesController < ApplicationController
-  def index
-  end
+  before_action :redirect_visitors, except: [:contact_show, :contact_create]
```

Comment:
Can we leave pages for all users (logged and not logged users) and create `DashboardController < ApplicationUserController` so we know that user is logged in (ApplicationUserController only needs a one line `before_action :authenticate_user!`)